### ISSUE TITLE
Vimeo Player stops responding to playing prop changes

### ISSUE THREAD

**Issue by thebrengun** [CookPete/react-player#520]
I'm using React Player with a project playing a series of Vimeo files and found an interesting edge case. The player stops responding to changes in the playing prop after playing one video and moving on to the next one but only if the videos have domain restricted embed settings. If the videos can be embedded anywhere there is no issue. But if they are restricted to a specific domain or domains the bug plays out like this:
  - First video is loaded, playing is true, the video plays to the end and onEnded is called, changing the video url. The play state can be affected by toggling the playing prop as expected.
  - The second video loads, playing is true, the video plays normally but does not start or stop based on user interactions that toggle playing prop.
Again, based a on a quick test with my videos, it seems to have something to do with the embed restrictions as this only happens with videos that have domain restricted embed settings.

**Comment by CookPete** [comment]
@thebrengun Do you have example URLs where this happens? Not sure how I can reproduce/debug this otherwise.

**Comment by thebrengun** [comment]
@CookPete Yes, I've set up a [repo](https://github.com/thebrengun/react-player-issue-520) to demonstrate the issue with a [live demo](https://thebrengun.github.io/react-player-issue-520/).

**Comment by CookPete** [comment]
@thebrengun Amazing work on the example repo/demo.

The crux of the problem is that the Vimeo SDK `play` event is _not called_ when playing the restricted videos, even thought the video _does play_:

https://github.com/CookPete/react-player/blob/db7bb5c63ca6063a4e344702ef9307474037c85c/src/players/Vimeo.js#L44-L47

This upsets the internal `isPlaying` flag and causes things to break.

https://github.com/CookPete/react-player/blob/db7bb5c63ca6063a4e344702ef9307474037c85c/src/Player.js#L155-L156

I'm not even sure why `https://www.vimeo.com/85876630` is working. It plays (without the `play` 
 event) if you set it as the initial `url` in the demo app, but loading it after the fact (via the custom URL field) gives an embed error: `“https://www.vimeo.com/85876630” is not embeddable.` which is.. correct behaviour?

One hacky workaround would be to manually play/pause the internal vimeo player when you set the `playing` state:

```js
handleToggleAllowPlaying(e) {
  this.setState(prevState => ({allowPlaying: !prevState.allowPlaying}));
  const vimeoPlayer = this.player.getInternalPlayer()
  if (vimeoPlayer) {
    if (this.state.allowPlaying) {
      vimeoPlayer.pause()
    } else {
      vimeoPlayer.play()
    }
  }
}

// ...

<ReactPlayer
  ref={player => { this.player = player }}
  // ...
/>
```

**Comment by thebrengun** [comment]
@CookPete Yes, thanks for looking into this!
